ee/fe/modules/500-openvpn/hooks/copy_rbd_secret.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2021 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package hooks
+
+import (
+	"errors"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
+
+	"github.com/deckhouse/deckhouse/go_lib/set"
+)
+
+type storageClassObject struct {
+	UserSecretName string
+}
+
+func filterStorageClass(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var sc storagev1.StorageClass
+
+	err := sdk.FromUnstructured(obj, &sc)
+	if err != nil {
+		return nil, err
+	}
+
+	var userSecretName string
+	if sc.Provisioner == "kubernetes.io/rbd" {
+		userSecretName = sc.Parameters["userSecretName"]
+		if userSecretName == "" {
+			return nil, errors.New("userSecretName for rbd StorageClass not found")
+		}
+	}
+
+	return storageClassObject{UserSecretName: userSecretName}, nil
+}
+
+func filterSecrets(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var secret v1.Secret
+	err := sdk.FromUnstructured(obj, &secret)
+	if err != nil {
+		return nil, err
+	}
+
+	return &secret, nil
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnAfterHelm: &go_hook.OrderedConfig{Order: 20},
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:                         "rbd_storageclass",
+			ApiVersion:                   "storage.k8s.io/v1",
+			Kind:                         "StorageClass",
+			ExecuteHookOnSynchronization: pointer.BoolPtr(false),
+			ExecuteHookOnEvents:          pointer.BoolPtr(false),
+			FilterFunc:                   filterStorageClass,
+		},
+		{
+			Name:                         "rbd_secret",
+			ApiVersion:                   "v1",
+			Kind:                         "Secret",
+			ExecuteHookOnSynchronization: pointer.BoolPtr(false),
+			ExecuteHookOnEvents:          pointer.BoolPtr(false),
+			FilterFunc:                   filterSecrets,
+			FieldSelector: &types.FieldSelector{
+				MatchExpressions: []types.FieldSelectorRequirement{
+					{
+						Field:    "type",
+						Operator: "Equals",
+						Value:    "kubernetes.io/rbd",
+					},
+				},
+			},
+		},
+	},
+}, copyRBDSecretHandler)
```

Comment:
There is the same hook in the `prometheus` module. It is required to move it to the go_lib then.